src/components/hosts/HostsTableCluster.tsx

```diff
@@ -0,0 +1,307 @@
+import { expandable, ICell, IRowData, sortable } from '@patternfly/react-table';
+import * as React from 'react';
+import { useDispatch } from 'react-redux';
+import {
+  Cluster,
+  deleteClusterHost,
+  disableClusterHost,
+  enableClusterHost,
+  getErrorMessage,
+  handleApiError,
+  Host,
+  installHost,
+  Inventory,
+  resetClusterHost,
+} from '../../api';
+import {
+  forceReload,
+  updateCluster,
+  updateHost,
+} from '../../features/clusters/currentClusterSlice';
+import { WithTestID } from '../../types';
+import { AlertsContext } from '../AlertsContextProvider';
+import {
+  HostsNotShowingLink,
+  HostsNotShowingLinkProps,
+} from '../clusterConfiguration/DiscoveryTroubleshootingModal';
+import HostsTable, { HostsTableProps } from './HostsTable';
+import { useModalDialogsContext } from './ModalDialogsContext';
+import { EmptyState } from '../ui/uiState';
+import { ConnectedIcon } from '@patternfly/react-icons';
+import { DiscoveryImageModalButton } from '../clusterConfiguration/discoveryImageModal';
+import {
+  canEditDisks as canEditDisksUtil,
+  canEditRole as canEditRoleUtil,
+  canInstallHost as canInstallHostUtil,
+  downloadHostInstallationLogs,
+  canEnable as canEnableUtil,
+  canDisable as canDisableUtil,
+  canDelete as canDeleteUtil,
+  canEditHost as canEditHostUtil,
+  canDownloadHostLogs,
+  canReset as canResetUtil,
+} from './utils';
+import EditHostModal from './EditHostModal';
+import { EventsModal } from '../ui/eventsModal';
+import ResetHostModal from './ResetHostModal';
+import DeleteHostModal from './DeleteHostModal';
+import { AdditionalNTPSourcesDialog } from './AdditionalNTPSourcesDialog';
+import HostsCount from './HostsCount';
+
+type HostsTableEmptyStateProps = {
+  cluster: Cluster;
+  setDiscoveryHintModalOpen?: HostsNotShowingLinkProps['setDiscoveryHintModalOpen'];
+};
+
+const HostsTableEmptyState: React.FC<HostsTableEmptyStateProps> = ({
+  cluster,
+  setDiscoveryHintModalOpen,
+}) => (
+  <EmptyState
+    icon={ConnectedIcon}
+    title="Waiting for hosts..."
+    content="Hosts may take a few minutes to appear here after booting."
+    primaryAction={<DiscoveryImageModalButton cluster={cluster} idPrefix="host-table-empty" />}
+    secondaryActions={
+      setDiscoveryHintModalOpen && [
+        <HostsNotShowingLink
+          key="hosts-not-showing"
+          setDiscoveryHintModalOpen={setDiscoveryHintModalOpen}
+        />,
+      ]
+    }
+  />
+);
+
+type HostsTableClusterProps = {
+  cluster: Cluster;
+  columns?: (string | ICell)[];
+  hostToHostTableRow?: HostsTableProps['hostToHostTableRow'];
+  skipDisabled?: boolean;
+  setDiscoveryHintModalOpen?: HostsNotShowingLinkProps['setDiscoveryHintModalOpen'];
+};
+
+const HostsTableCluster: React.FC<HostsTableClusterProps & WithTestID> = ({
```

Comment:
Nit: `ClusterHostsTable` might be a bit better.

Reply:
sounds good!
I also wonder  - dont we want to move these generic components to some specific folder so we can have easier time identifying if its being reused in ACM/OCP or not ?

Reply:
I would not organize that for ACM/OCP. We are a library which is used in other contexts as well (+3).

We have already "naming contexts" on the top-level export. We can put some effort to better identify what to export/hide, recently we blindly reexport almost everything.